### DISCUSSION TITLE
Block List/Grid - Use HTML-files from Razor Class Library as templates

### DISCUSSION THREAD

**Discussion by enkelmedia** [umbraco/Umbraco-CMS#13555] [Features and ideas]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.0.0

### Bug summary

Now that support putting package.manifest files inside a Razor Class Library has been added it would also be nice to be able to choose files from the library as view templates for e.g the Block List or Block Grid Editor.

### Specifics

We have a "common" library that we import in all our projects `Obviuse.Umbraco`, this is a Razor Class Library with some common views and extensions for the backoffice.

At the moment we use the "old" approach having a build-target in the library that will copy any files from App_Plugins into the web-project. But after the support for adding static content and package.manifest to a library was added I was thinking that I could put some of the html-files used for block list or block grid in our RCL.

The files in the RCL is accessible from the browser but there is now way to see them in the view-picker

![image](https://user-images.githubusercontent.com/1782524/206687676-82c74d1c-b72c-4796-93f1-fe8805073f53.png)

I'm proposing that we should either list files from RCLs in this picker or make it possible to manually configure the path to the view (as far as I can tell it's not possible).

### Steps to reproduce

Create a new Umbraco 11 site.
Create a RCL with some html-files in `App_Plugins` in the wwwroot-folder.
Open the view picker and see that the view does not appear.

### Expected result / actual result

Files from RCLs should be listed and or workaround with manual entering of the view path.

**Comment by nikolajlauridsen** [comment]
Hey @enkelmedia that sounds very interesting, and possibly like a great feature, for now, I'll go ahead and convert this into a discussion since it's really more of a feature request 😄
